models/User.js

```diff
@@ -431,13 +439,32 @@ userSchema.methods.unitLoss = function(lossPercentage) {
 	});
 
 	// Remove hp from your hero depending on the loss percentage
-	this.hero.currentHealth = Math.floor(this.hero.currentHealth * lossPercentage);
+	this.hero.currentHealth = this.hero.currentHealth - Math.floor(this.hero.currentHealth * lossPercentage);
 
+	// if the hero died
+	if (this.hero.currentHealth <= 0 && this.hero.rank > 0) {
+		this.hero.rank -= 1;
+		this.hero.expToNextRank = heroExpToNextLevel[this.hero.rank];
+	}
+	// failsafe to ensure that hero doesn't have negative hp
+
+	if (this.hero.currentHealth < 0) {
```

Comment:
Det skal strengt talt ikke være nødvendig siden den eneste muligheten for at Math.floor(this.hero.currenHealth * lossPercentage) blir negativt er hvis currentHealth allerede er negativ eller hvis lossPercentage er negativ.

Hvis du er keen på å ha en failsafe så foreslår jeg å sette en: if lossPercentage < 0 || this.hero.currentHealth < 0, helt i begynnelsen av funksjonen